pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -944,6 +978,28 @@ func (c *Controller) WorkloadInstanceHandler(si *model.WorkloadInstance, event m
 	}
 }
 
+func (c *Controller) onNamespaceEvent(obj interface{}, ev model.Event) error {
+	var nw string
+	if ev != model.EventDelete {
+		ns, ok := obj.(*v1.Namespace)
+		if !ok {
+			log.Warnf("Namespace watch getting wrong type in event: %T", obj)
+			return nil
+		}
+		nw = ns.Labels[label.IstioNetwork]
```

Comment:
I am not sure which component will label this network

Reply:
Check if the label exists

Reply:
If the label doesn't exist, it's empty string and we unset.

Reply:
The first question, it requires user to set the ns label, right?

Reply:
Yes, that's the recommended route. We _could_ also set it if istioctl creates istio-system.